src/test/java/org/mockitousage/matchers/VarargsTest.java

```diff
@@ -342,8 +329,281 @@ public void shouldNotMatchVaraArgs() {
+    }
+
+    @Test
+    public void shouldMockVarargsInvocationForSuperType() {
+        given(mock.varargsReturningString(any(Object[].class))).willReturn("a");
+
+        assertThat(mock.varargsReturningString("a", "b")).isEqualTo("a");
+    }
+
+    @Test
+    public void shouldHandleArrayVarargsMethods() {
+        given(mock.arrayVarargsMethod(any(String[][].class))).willReturn(1);
+
+        assertThat(mock.arrayVarargsMethod(new String[] {})).isEqualTo(1);
+    }
+
+    @Test
+    public void shouldCaptureVarArgs_NullArrayArg1() {
+        mock.varargs((String[]) null);
+        ArgumentCaptor<String[]> captor = ArgumentCaptor.forClass(String[].class);
+
+        verify(mock).varargs(captor.capture());
+
+        assertThat(captor.getValue()).isNull();
+    }
+
+    @Test
+    public void shouldCaptureVarArgs_NullArrayArg2() {
+        mock.varargs((String[]) null);
+        ArgumentCaptor<String> captor = ArgumentCaptor.forClass(String.class);
+
+        verify(mock).varargs(captor.capture());
+
+        assertThat(captor.getValue()).isNull();
+    }
+
+    @Test
+    public void shouldVerifyVarArgs_any_NullArrayArg1() {
+        mock.varargs((String[]) null);
+
+        verify(mock).varargs(any());
+    }
+
+    @Test
+    public void shouldVerifyVarArgs_any_NullArrayArg2() {
+        mock.varargs((String) null);
+
+        verify(mock).varargs(any());
+    }
+
+    @Test
+    public void shouldVerifyVarArgs_eq_NullArrayArg1() {
+        mock.varargs((String[]) null);
+
+        verify(mock).varargs(eq(null));
+    }
+
+    @Test
+    public void shouldVerifyVarArgs_eq_NullArrayArg2() {
+        mock.varargs((String) null);
+
+        verify(mock).varargs(eq(null));
+    }
+
+    @Test
+    public void shouldVerifyVarArgs_isNull_NullArrayArg() {
+        mock.varargs((String) null);
+
+        verify(mock).varargs(isNull(String.class));
+    }
+
+    @Test
+    public void shouldVerifyVarArgs_isNull_NullArrayArg2() {
+        mock.varargs((String) null);
+
+        verify(mock).varargs(isNull());
+    }
+
+    @Test
+    public void shouldVerifyExactlyOneVarArg_isA() {
+        mock.varargs("one param");
+
+        verify(mock).varargs(isA(String.class));
+    }
+
+    @Test
+    public void shouldNotVerifyExactlyOneVarArg_isA() {
+        mock.varargs("two", "params");
+
+        verify(mock, never()).varargs(isA(String.class));
+    }
+
+    @Test
+    public void shouldVerifyVarArgArray_isA() {
+        mock.varargs("one param");
+
+        verify(mock).varargs(isA(String[].class));
+    }
+
+    @Test
+    public void shouldVerifyVarArgArray_isA2() {
+        mock.varargs("two", "params");
+
+        verify(mock).varargs(isA(String[].class));
+    }
+
+    @Test
+    public void shouldVerifyExactlyOneVarArg_any() {
+        mock.varargs("one param");
+
+        verify(mock).varargs(any(String.class));
+    }
+
+    @Test
+    @Ignore("Fails due to https://github.com/mockito/mockito/issues/1593")
```

Comment:
Your PR description claims this is fixed. Can we unignore this test?

Reply:
Ah, good catch. #1593 isn't resolved until we remove the broken `VarargMatcher` "use the last vararg matcher to match all additional parameters".  I've remove this from the fixed list.